qiskit/providers/backend_compat.py

```diff
@@ -13,24 +13,367 @@
+        else:
+            logger.warning(
+                "Definition of instruction %s is not found in the Qiskit namespace and "
+                "GateConfig is not provided by the BackendConfiguration payload. "
+                "Qiskit Gate model cannot be instantiated for this instruction and "
+                "this instruction is silently excluded from the Target. "
+                "Please add new gate class to Qiskit or provide GateConfig for this name.",
+                name,
+            )
+            unsupported_instructions.append(name)
+
+    for name in unsupported_instructions:
+        all_instructions.remove(name)
+
+    # Create empty inst properties from gate configs
+    for name, spec in gate_configs.items():
+        if hasattr(spec, "coupling_map"):
+            coupling_map = spec.coupling_map
+            prop_name_map[name] = dict.fromkeys(map(tuple, coupling_map))
+        else:
+            prop_name_map[name] = None
+
+    # Populate instruction properties
+    if properties:
+        qubit_properties = list(map(_decode_qubit_property, properties["qubits"]))
+        in_data["qubit_properties"] = qubit_properties
+        faulty_qubits = {q for q, prop in enumerate(qubit_properties) if not prop.operational}
+
+        for gate_spec in map(GateSchema.from_dict, properties["gates"]):
+            name = gate_spec.gate
+            qubits = tuple(gate_spec.qubits)
+            if name not in all_instructions:
+                logger.info(
+                    "Gate property for instruction %s on qubits %s is found "
+                    "in the BackendProperties payload. However, this gate is not included in the "
+                    "basis_gates or supported_instructions, or maybe the gate model "
+                    "is not defined in the Qiskit namespace. This gate is ignored.",
+                    name,
+                    qubits,
+                )
+                continue
+            inst_prop, operational = _decode_instruction_property(gate_spec)
+            if set.intersection(faulty_qubits, qubits) or not operational:
+                faulty_ops.add((name, qubits))
+                try:
+                    del prop_name_map[name][qubits]
+                except KeyError:
+                    pass
+                continue
+            if prop_name_map[name] is None:
+                prop_name_map[name] = {}
+            prop_name_map[name][qubits] = inst_prop
+        # Measure instruction property is stored in qubit property in IBM
+        measure_props = list(map(_decode_measure_property, properties["qubits"]))
+        prop_name_map["measure"] = {}
+        for qubit, measure_prop in enumerate(measure_props):
+            if qubit in faulty_qubits:
+                continue
+            qubits = (qubit,)
+            prop_name_map["measure"][qubits] = measure_prop
+
+    # Special case for real IBM backend. They don't have delay in gate configuration.
+    if "delay" not in prop_name_map:
```

Comment:
```suggestion
    if add_delay and  "delay" not in prop_name_map:
```
e.g.) Regarding `add_delay`, adding `add_delay` to this if-statement is enough to cover the same functionality as the legacy one.

Reply:
I didn't think of adding `add_delay` here because I was a bit skeptical if this is even needed here, being compatible with the legacy one was not in my mind as I have already said, only `target` of `FakeBackendV2` used `convert_to_target`, I was more geared towards changing the usage of `convert_to_target` in `BackendV2Converter` as well.
I have applied the suggestion :)